core/appeal/service.go

```diff
@@ -486,6 +493,367 @@ func validateAppealOnBehalf(a *domain.Appeal, policy *domain.Policy) error {
 	return nil
 }
 
+// Patch record
+func (s *Service) Patch(ctx context.Context, appeal *domain.Appeal, opts ...CreateAppealOption) error {
+	existingAppeal, err := s.GetByID(ctx, appeal.ID)
+	if err != nil {
+		return fmt.Errorf("error getting existing appeal: %w", err)
+	}
+
+	if existingAppeal.Status != domain.AppealStatusPending {
+		return ErrAppealStatusInvalid
+	}
+
+	createAppealOpts := &createAppealOptions{}
+	for _, opt := range opts {
+		opt(createAppealOpts)
+	}
+	isAdditionalAppealCreation := createAppealOpts.IsAdditionalAppeal
```

Comment:
this flag shouldn't be relevant for edit appeal. we can remove this and remove options from `Patch` method param.

here in `Patch` we are already calling `GrantAccessToProvider()` (in auto-approval case) method which will handle the grant creation, grant provision, and additional appeals (if any) the same way as appeal creation did